gpt4all-bindings/typescript/src/util.js

```diff
@@ -69,65 +87,66 @@ function downloadModel(modelName, options = {}) {
     const abortController = new AbortController();
     const signal = abortController.signal;
 
-    // wrapper function to get the readable stream from request
-    const fetchModel = (fetchOpts = {}) =>
+    // a promise that executes and writes to a stream. Resolves to the path the model was downloaded to when done writing.
+    const downloadPromise = new Promise((resolve, reject) => {
+        const writeStream = createWriteStream(
+            partialModelPath,
+            writeStreamOpts
+        );
+
         fetch(modelUrl, {
             signal,
-            ...fetchOpts,
-        }).then((res) => {
-            if (!res.ok) {
-                throw Error(
-                    `Failed to download model from ${modelUrl} - ${res.statusText}`
-                );
-            }
-            return res.body.getReader();
-        });
-
-    // a promise that executes and writes to a stream. Resolves when done writing.
-    const res = new Promise((resolve, reject) => {
-        fetchModel({ headers })
-            // Resolves an array of a reader and writestream.
-            .then((reader) => [
-                reader,
-                createWriteStream(partialModelPath, writeStreamOpts),
-            ])
-            .then(async ([readable, wstream]) => {
+            headers,
+        })
+            .then((res) => {
+                if (!res.ok) {
+                    reject(
+                        Error(
```

Comment:
im starting to hate this absurd thing that prettier does haha we might need to switch or alter rules